mm2src/coins/eth/eth_swap_v2/eth_maker_swap_v2.rs

```diff
@@ -0,0 +1,523 @@
+use super::{validate_from_to_and_status, validate_payment_args, EthPaymentType, PrepareTxDataError, ZERO_VALUE};
+use crate::coin_errors::{ValidatePaymentError, ValidatePaymentResult};
+use crate::eth::{decode_contract_call, get_function_input_data, wei_from_big_decimal, EthCoin, EthCoinType,
+                 MakerPaymentStateV2, SignedEthTx, MAKER_SWAP_V2};
+use crate::{ParseCoinAssocTypes, RefundMakerPaymentSecretArgs, RefundMakerPaymentTimelockArgs, SendMakerPaymentArgs,
+            SpendMakerPaymentArgs, SwapTxTypeWithSecretHash, TransactionErr, ValidateMakerPaymentArgs,
+            WaitForPaymentSpendError};
+use common::executor::Timer;
+use common::now_sec;
+use ethabi::{Function, Token};
+use ethcore_transaction::Action;
+use ethereum_types::{Address, Public, U256};
+use ethkey::public_to_address;
+use futures::compat::Future01CompatExt;
+use mm2_err_handle::mm_error::MmError;
+use mm2_err_handle::prelude::{MapToMmResult, MmResult};
+use std::convert::TryInto;
+use web3::types::TransactionId;
+
+const ETH_MAKER_PAYMENT: &str = "ethMakerPayment";
+const ERC20_MAKER_PAYMENT: &str = "erc20MakerPayment";
+
+struct MakerPaymentArgs {
+    taker_address: Address,
+    taker_secret_hash: [u8; 32],
```

Comment:
Did you think of using in TPU 'new type' pattern, like SwapSecret([u8; 32]), instead of simple types?

Reply:
I think, it could be useful in large systems, while ETH coin specific fixed size secret hashes / secrets are used just in inner Eth tpu swap_v2 implementations. Additionally, we use them in straightforward ways (to provide fixed bytes). I dont think the abstraction type may offer much practical benefit in our case.
plus for each new type, we will likely need to implement traits like Clone, Copy, Debug, etc, which could increase the amount of boilerplate code.

But I will keep this design approach in mind.